wire-library/wire-runtime-swift/src/main/swift/propertyWrappers/StringEncoded+Data.swift

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+
+extension Data : StringCodable {
+    public func stringEncodedValue(in encoder: Encoder) throws -> String {
+        switch encoder.stringEncodedDataEncodingStrategy {
+        case .base64:
+            return base64EncodedString()
+
+        case .base64url:
+            return base64urlEncodedString()
+        }
+    }
+
+    public init(encodedValue: String, from decoder: Decoder) throws {
+        let possibleValue = Data(base64Encoded: encodedValue) ??
+            Data(base64urlEncoded: encodedValue)
+
+        guard let value = possibleValue else {
+            throw ProtoDecoder.Error.unparsableString(
+                type: Data.self,
+                value: encodedValue
+            )
+        }
+        self = value
+    }
+}
+
+extension Data : EmptyInitializable {}
+
+public extension Data {
+    func base64urlEncodedData(options: Data.Base64EncodingOptions = []) -> Data {
+        let string = base64urlEncodedString(options: options)
+        return Data(string.utf8)
+    }
+
+    func base64urlEncodedString(options: Data.Base64EncodingOptions = []) -> String {
+        let base64 = base64EncodedString(options: options)
+        return base64
+            .replacingOccurrences(of: "/", with: "_")
+            .replacingOccurrences(of: "+", with: "-")
+            .replacingOccurrences(of: "=", with: "")
+    }
+
+    init?(
+        base64urlEncoded base64urlData: Data,
```

Comment:
nit: I think this should technically be named `base64URLEncoded base64URLData:`

Reply:
Yeah I went back and forth on this a few times.
I went with this originally because of the [RFC](https://www.rfc-editor.org/rfc/rfc4648#section-5)

>  This encoding may be referred to as "base64url".  This encoding
   should not be regarded as the same as the "base64" encoding and
   should not be referred to as only "base64".  Unless clarified
   otherwise, "base64" refers to the base 64 in the previous section.

but semantically I agree, it feels like it should be `URL`

Reply:
(renamed)